app/services/apply_api/retrieve_applications.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+module ApplyApi
+  class RetrieveApplications
+    include ServicePattern
+
+    SUCCESS = 200
+
+    class HttpError < StandardError; end
+
+    def initialize(accepted_since:)
+      @accepted_since = accepted_since
+    end
+
+    def call
+      log_request!
+
+      if response.code != SUCCESS
+        raise HttpError, "status: #{response.code}, body: #{response.body}, headers: #{response.headers}"
+      end
+
+      accepted_since.nil? ? all_applications : filtered_applications
+    end
+
+  private
+
+    attr_reader :accepted_since
+
+    def filtered_applications
+      all_applications.select do |a|
+        Time.zone.parse(a["attributes"]["offer"]["offer_accepted_at"]) >= accepted_since
+      end
+    end
+
+    def all_applications
+      @all_applications ||= JSON(response.body)
+    end
+
+    # TODO: Swap to Client.get("URL") once we have the Apply endpoint details.
```

Comment:
Will the new endpoint allow us to provide an accepted_since attribute? Then we wouldn't need to have to have this `filtered_applications` method.

Reply:
I believe we've agreed that it will just return all the applications for the year, since there won't be a huge number. But @gpeng / @duncanjbrown will be able to correct me if I'm wrong!

I initially assumed that the endpoint would support a `accepted_since` attribute.

Reply:
If it keeps things much simpler on this side it's not much work to add it. (cc @cwrw and @AgaDufrat, who are building it!)

Q tho: is acceptance date the right criterion — don't we want `recruited_at`?

Reply:
We have currently not included the `offer` object in the [API implementation](https://github.com/DFE-Digital/apply-for-teacher-training/pull/4380) on our side so access to the `offer_accepted_at` is not supported. We are happy to include it if the attributes under the [offer object](https://www.apply-for-teacher-training.service.gov.uk/api-docs/reference#offer-object) are required for the integration.

Having had a look at `offer_accepted_at` attribute which is taken from the application choice `accepted_at` attribute, it seems `recruited_at` might be more suited. The `recruited_at` attribute is set when confirming the application offer conditions or for an unconditional offer, as well as reinstating an offer after deferral, all transitions to the recruited state. 
The `accepted_at` value is set when an offer is accepted by the candidate and transitions into pending conditions state which might not be suitable here.

In terms of the parameters we support on the API, we have currently included a `recruitment_cycle_year` parameter for the API, but happy to include another one to filter by the `recruited` or `accepted` since value depending on the criteria required. I actually prefer including those to protect against performance issues we might encounter later, as for now we only have around 260 applications in total in the recruited state, however that might change later and would be good to somehow pull those in incrementally to avoid timeouts. Let me know your thoughts

Reply:
Thanks for clarifying the date stuff! I didn't give much thought to picking `offer_accepted_at ` - I needed a timestamp to flesh out the architecture and was waiting to confirm 👍  Will change this to `recruited_at`.

It would mean less code for us if we could provide a `recruited_since` parameter to filter the applications to just the ones we needed. @gpeng - I can't quite remember whether there was a separate reason why it could be useful for us to receive all of the applications?

Reply:
I was also wondering around providers and maybe filtering by their codes in a parameter, to help remove the need to check if they exist on your end as well. I scheduled a quick chat later today as might be easier to iron out those questions together